### PR TITLE
Fix mixed_text metadata fields without collection

### PR DESCRIPTION
**Pull request by KatrinIhler**
We have disabled the listproviders for some mixed_text fields in the episode and series metadata for performance reasons, meaning that those fields don't have a collection, which is a valid config for the old Admin UI. However, this currently breaks the new one:
image

Simply checking if field.collection is defined before checking its length should fix this.

Warning: I didn't test this because I didn't want to deal with figuring out how to do this against my local Opencast instance. But this is simple enough, so it should be fine...right?

Edit: I'm also unsure which branch to point this at to get it into Opencast as soon as possible, maybe you can advise me on that?

### CODE REVIEW

**Comment by Arnei** [comment]
Could you share what config(s) you used to trigger the error?

Are the performance issues caused by the new admin ui? Want to open an issue for that?

> Edit: I'm also unsure which branch to point this at to get it into Opencast as soon as possible, maybe you can advise me on that?

You chose the correct branch. There is also no fixed release schedule atm, you can basically cut a release whenever.

**Comment by KatrinIhler** [comment]
> Could you share what config(s) you used to trigger the error?

Sure! Just comment out the listprovider for at least one field with type `mixed_text` in either `org.opencastproject.ui.metadata.CatalogUIAdapterFactory-episode-common.cfg` or `org.opencastproject.ui.metadata.CatalogUIAdapterFactory-series-common.cfg` to trigger the error for events and series details respectively. Example:

```
# Presenters
property.presenters.inputID=creator
property.presenters.label=EVENTS.EVENTS.DETAILS.METADATA.PRESENTERS
property.presenters.type=mixed_text
property.presenters.readOnly=false
property.presenters.required=false
# property.presenters.listprovider=CONTRIBUTORS
property.presenters.order=7
```

> Are the performance issues caused by the new admin ui? Want to open an issue for that?

Nope, that's Opencast's fault, and it's a known problem with requesting all users.

**Comment by KatrinIhler** [comment]
Oh, and I also just realized I didn't describe the error correctly: If you open either series or event details, the whole Admin UI turns white. :D

**Review by Arnei — CHANGES_REQUESTED** [review]
Makes sense to me and works. Below is a suggestion that should be functionally equivalent, but less code :D

File: app/src/components/events/partials/ModalTabsAndPages/DetailsMetadataTab.tsx [hunk 1]
```diff
@@ -129,6 +129,7 @@ const DetailsMetadataTab = ({
 																<td className="editable">
 																	{/* Render single value or multi value editable input */}
 																	{field.type === "mixed_text" &&
+																	field.collection &&
 																	field.collection.length !== 0 ? (
```
[the remark this answers is not in the document]
Comment (KatrinIhler) [reply app/src/components/events/partials/ModalTabsAndPages/DetailsMetadataTab.tsx]: Julian already suggested this to me, but I didn't dare try that without testing. But if you say so... ;)

**Review by Arnei — APPROVED** [review]

**Comment by Arnei** [comment]
Actually wait a sec, the same error occurs in `NewMetadataPage.tsx` (line 44). Other Metadata related tsx files seems to be fine.

**Comment by KatrinIhler** [comment]
> Actually wait a sec, the same error occurs in `NewMetadataPage.tsx` (line 44). Other Metadata related tsx files seems to be fine.

Good catch. Actually affects the extended metadata as well. But now I think I got everything. :D Again: I didn't test this, especially with the new syntax using ?.

**Review by Arnei — APPROVED** [review]

**Pull request merged**